install.sh

```diff
@@ -22,86 +93,332 @@ deps=(
 "librtaudio6"
 "libkf5bluezqt-dev"
 "libtag1-dev"
+"qml-module-qtquick2"
+"libglib2.0-dev"
+"libgstreamer1.0-dev"
+"gstreamer1.0-plugins-base-apps"
+"gstreamer1.0-plugins-bad"
+"gstreamer1.0-libav"
+"gstreamer1.0-alsa"
+"libgstreamer-plugins-base1.0-dev"
+"qtdeclarative5-dev"
+"qt5-default"
+"libgstreamer-plugins-bad1.0-dev"
+"libunwind-dev"
+"qml-module-qtmultimedia"
 )
 
-#loop through dependencies and install
-for app in ${deps[@]}; do
-	echo "installing: " $app
-	sudo apt -qq -o=Dpkg::Use-Pty=0 install $app -y > /dev/null 2> /dev/null
-
-	if [[ $? > 0 ]]
-	then
-	    echo $app " Failed to install, quitting"
-	    exit
-	else
-	    echo $app " Installed ok"
-	    echo
-	    
-	fi
-done
-
-echo "All dependencies installed"
-echo
-
-#make ilclient
-echo "making ilclient"
-make /opt/vc/src/hello_pi/libs/ilclient
-if [[ $? > 0 ]]
+###############################  dependencies  #########################
+if [ $deps = false ]
   then
-    echo "unable to make ilclient"
-  exit
-else
-  echo "made ok"
-  echo
+    echo skipping dependencies '\n'
+  else
+    #loop through dependencies and install
+    echo installing dependencies
+    echo Running apt update
+    sudo apt update
+
+    installString="sudo apt install -y "
+
+    #create apt install string
+    for i in ${dependencies[@]}; do
+      installString+=" $i"
+    done
+
+    #run install
+    ${installString}
+    if [[ $? -eq 0 ]]; then
+        echo -e All dependencies Installed ok '\n'
+    else
+        echo Package failed to install with error code $?, quitting check logs above
+        exit
+    fi
 fi
 
-#begin cmake
-echo "beginning cmake for raspberry pi"
-cmake -DRPI_BUILD=TRUE -DGST_BUILD=TRUE -DCMAKE_BUILD_TYPE=Release . 
-if [[ $? > 0 ]]
-  then
-    echo "Cmake error"
-  exit
+
+###############################  AASDK #########################
+if [ $aasdk = false ]; then
+	echo -e Skipping aasdk '\n'
 else
-  echo "Cmake ok"
-  echo #####
-fi
+  #change to parent directory
+  cd ..
 
-#start make
-echo "beginning make, this will take a while"
-make
-if [[ $? > 0 ]]
-  then
-    echo "make error check output above"
+  #clone aasdk
+  git clone $aasdkRepo
+  if [[ $? -eq 0 ]]; then
+    echo -e Aasdk Cloned ok '\n'
+  else
+    cd aasdk
+    if [[ $? -eq 0 ]]; then
+      git pull $aasdkRepo
+      echo -e Aasdk Cloned OK '\n'
+      cd ..
+    else
+      echo Aasdk clone/pull error
+      exit
+    fi
+  fi
+
+  #change into aasdk folder
+  echo -e moving to aasdk '\n'
+  cd aasdk
+
+  #beginning cmake
+  cmake -DCMAKE_BUILD_TYPE=Release .
```

Comment:
would we also want to create a `build` directory and do the build there? Just in case something fails the entire build can just be wiped and retired (same for openauto)